crates/arrow_sql_gen/src/statement.rs

```diff
@@ -308,33 +317,35 @@ impl InsertBuilder {
                             }
                         }
                     }
-                    _ => unimplemented!(
-                        "Data type mapping not implemented for {}",
-                        column.data_type()
-                    ),
+                    _ => return Result::Err(Error::FailedToCreateInsertStatement { source: format!("Data type mapping not implemented for {}", column.data_type()).into() }) 
```

Comment:
This should be a separate Snafu error type specific to Unimplemented

Reply:
Done